### PR TITLE
refactor: move helper functions under common, in usage specific named files

### PR DESCRIPTION
**Pull request by alexstan12**
**What type of PR is this?**
cleanup

**Which issue does this PR fix**:
moves helper functions and their dependencies under the common package.
creates a new package under api called api/errors for evading circular imports related to importing error variables

**What does this PR do / Why do we need it**:

**If an issue # is not available please add repro steps and logs showing the issue**:

**Testing done on this change**:

**Automation added to e2e**:

**Will this break upgrades or downgrades?**

**Does this PR introduce any user-facing change?**:

```release-note

```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/api/routes.go [hunk 1]
```diff
@@ -38,7 +38,7 @@ import (
 	"zotregistry.io/zot/pkg/meta"
 	zreg "zotregistry.io/zot/pkg/regexp"
 	localCtx "zotregistry.io/zot/pkg/requestcontext"
-	storageCommon "zotregistry.io/zot/pkg/storage/common"
+	storagecommon "zotregistry.io/zot/pkg/storage/common"
```
Comment (andaaron) [inline pkg/api/routes.go:41]: Why the name change?

Comment (alexstan12) [reply pkg/api/routes.go:41]: formatting error, will revert

File: pkg/api/errors_test.go
[the remark this answers is not in the document]
Comment (alexstan12) [reply pkg/api/errors_test.go:1]: done

**Review by andaaron — DISMISSED** [review]

File: pkg/common/references.go
Comment (andaaron) [inline pkg/common/references.go:1]: Rename this to oci.go and move the stuff from current oci.go back here.

**Review by andaaron — APPROVED** [review]

**Pull request merged**